# DNA Sequence Alignment

## Overview

This project implements a DNA sequence alignment algorithm using dynamic programming techniques. The goal is to align two DNA sequences effectively, allowing for optimal matches while accounting for potential mismatches and gaps. The algorithm generates a score matrix, fills it according to defined scoring criteria, and performs backtracking to find the best alignment.

## Features

- **Score Matrix Generation**: Constructs a score matrix to evaluate the alignment between two sequences.
- **Scoring Phase**: Utilizes dynamic programming to fill the score matrix based on match/mismatch and gap penalties.
- **Backtracking**: Retrieves the optimal alignment of the sequences by backtracking through the filled score matrix.
- **CSV Input**: Reads DNA sequences from a CSV file, making it easy to align multiple pairs of sequences at once.
- **Flexible Scoring**: Customizable scoring system for matches, mismatches, and gap penalties.

## Applications

This tool can be applied in various fields of bioinformatics, including:
- Comparative genomics
- Phylogenetics
- Evolutionary biology
- Genetic variation analysis

## Installation

1. Clone the repository:

   ```bash
   git clone https://github.com/Yamil-Serrano/DNA-Sequence-Alignment.git
   
## License

This project is licensed under the [Creative Commons Attribution-NonCommercial 4.0 International License](LICENSE.md).

## Contact

If you have any questions or suggestions, feel free to reach out to me:

GitHub: [Neowizen](https://github.com/Yamil-Serrano)
